### ISSUE TITLE
Enhance the enable binding

### ISSUE THREAD

**Issue by cervengoc** [knockout/knockout#2107]
I needed to enable/disable links based on viewmodel, but the current implementation of `enable` binding works properly only in input tags. So first of all I would like to have better support for link elements.

My proposal

1. When disabled, add the `disabled` attribute as well, so the element can be styled according to it.
1/b. When disabled, add a `"disabled"` class to the element, so it can be styled. Also, CSS frameworks like `bootstrap` make use of this class.
2. In the `init` handler of `enable` binding, register a `click` event listener if the element is a link, and in the listener `return false` if the element is disabled, so the `click` event is prevented at the very early stage. Perhaps similar handlers for some key events would be nice too.
3. Set the `after` property of `event` and `click` binding handlers to ensure that they run after `enable` and `disable` binding.

**Comment by cervengoc** [comment]
There is an alternative way to disable links completely (even the `title` won't show up on mouse over).

```
a[disabled] { // or a.disabled
  pointer-events: none;
}
```

Of course this has some side effects, but should be acceptable in most cases. 

So it seems like as far as clicking goes, it can be solved outside KO. Still, it would be needed to add the `disabled` attribute, or the `disabled` class, or actually the best would be both IMO.

**Comment by brianmhunt** [comment]
Since everyone is going to have different behaviours depending on whether they use a framework or their design choices, and in the future all our expectations may differ on the correct behaviour, I think the correct thing to do here would be to have a custom binding handler that combines the `attr`, `css` and `event`/`click` bindings in the order that the user expects.

In other words, I think in the core the bindings ought to be "atomic" – do one thing well – and it's up to the user to create combinations of bindings.

This would be a great example of a custom binding.

Does that make sense? :)

**Comment by cervengoc** [comment]
Yes it does make sense, that's why I reduced my idea to just add the `disabled` attribute beside changing the `disabled` property. That should be enough for proper styling of anything disabled.

Is there any reason why the `enable` binding removes the `disabled` attribute, but never adds it?

**Comment by brianmhunt** [comment]
Thanks for clarifying :) 

I think adding a truthy property added it as an attribute (but setting it to falsy doesn't remove it in old IE).  But that's just recollection – answer being: I think it's a legacy issue.

**Comment by cervengoc** [comment]
The browsers adds the attribute only on input elements as far as I know. Perhaps on `a` elements it's not even valid HTML, I have to check. That's why I've suggested the `disabled` class, which sounds quite framework independent IMHO.

**Comment by cervengoc** [comment]
@mbest What do you think about a possible merge if I create a PR? I mean the implementation which would add the `disabled` attribute regardless of what the element is.

**Comment by mbest** [comment]
Although I understand the desire to style "disabled" elements, I think that's beyond the score of the `disable` binding.

**Comment by cervengoc** [comment]
Styling is just one aspect of this, in general I'd like to achieve that `enabled` binding should make sens for any element, not just for inputs. For example why couldn't I enable/disable a `div` element where one can drag things. In that case I would consume this information via JS somewhere.

As far as I know, adding attributes to elements which they don't consume doesn't violate standards, or does it? So I mean it shouldn't hurt anyone to add a `disabled` attribute to any element.

**Comment by brianmhunt** [comment]
FWIW I have been mulling various "composite" bindings i.e. (re-)combinations of existing bindings for tko.  I've not landed on a candidate design, but I feel like this is the exact use case.

**Comment by codymullins** [comment]
I agree with @cervengoc. Links are a pretty common thing that need to be disabled.
